reference/FungibleToken.scilla

```diff
@@ -294,105 +269,171 @@ transition AuthorizeOperator(operator: ByStr20)
+    current_allowance =
+      match get_current_allowance with
+      | Some allowance => allowance
+      | None => zero
+      end;
+    new_allowance = builtin add current_allowance amount;
+    allowances_map[_sender][spender] := new_allowance;
+    e = {_eventname : "IncreasedAllowance"; sender : _sender; spender: spender; new_allowance : new_allowance};
+    event e
   end
 end
 
-(* @dev: Emits an event with the number of tokens an approved_spender is allowed to spend on behalf of the token_owner. *)
-(* param token_owner:  Address of a token_owner.                                                                        *)
-(* param spender:      Address of the approved_spender.                                                                 *)
-transition Allowance(token_owner: ByStr20, spender: ByStr20)
-  get_allowance <- allowances_map[token_owner][spender];
-  match get_allowance with
-  | None =>
-    (* approved_spender does not exists *)
-    err = CodeNotFound;
-    IssueErrorEvent err
-  | Some val =>
-    e = {_eventname : "Allowance"; token_owner: token_owner; spender : spender;  allowance : val};
+(* @dev: Decrease the allowance of an approved_spender over the caller’s tokens. Only token_owner allowed to invoke. *)
+(* param spender:      Address of the designated approved_spender.                                                   *)
+(* param amount:       Number of tokens to be decreased as allowance for the approved_spender.                       *)
+transition decreaseAllowance(spender: ByStr20, amount: Uint128)
+  (* Checks if the _sender and approved_spender is the same *)
+  is_owner = builtin eq _sender spender;
+  match is_owner with
+  | True =>
+    err = CodeNotAuthorized;
+    issueErrorEvent err
+  | False =>
+    get_current_allowance <- allowances_map[_sender][spender];
+    current_allowance =
+      match get_current_allowance with
+      | Some allowance => allowance
+      | None => zero
+      end;
+    new_allowance =
+      let amount_le_allowance = uint128_le amount current_allowance in
+      match amount_le_allowance with
+      | True => builtin sub current_allowance amount
+      | False => zero
+      end;
+    allowances_map[_sender][spender] := new_allowance;
+    e = {_eventname : "DecreasedAllowance"; sender : _sender; spender: spender; new_allowance : new_allowance};
     event e
   end
 end
 
+(* @dev: Moves an amount tokens from _sender to the recipient. Used by token_owner. *)
+(* @dev: Balance of recipient will increase. Balance of _sender will decrease.      *)
+(* @param to:  Address of the recipient whose balance is increased.                 *)
+(* @param amount:     Amount of tokens to be sent.                                  *)
+transition transfer(to: ByStr20, amount: Uint128)
+  authorizedMoveIfSufficientBalance _sender to amount;
+  e = {_eventname : "TransferSuccess"; sender : _sender; recipient : to; amount : amount};
```

Comment:
XSGD user the `_eventname : Transfer`

Reply:
Should we stick to style guide? I understand that the eventname is to be picked up by the app, and it will be good to have some consistency. If we think the style guide isn't accurate, we should change it.

Reply:
I believe it would be easier to enforce a style by implementing a tool for the task. It could be a small internship project.

Reply:
Will be shifting every transition to PascalCase.